rust/src/types.rs

```diff
@@ -3240,6 +3458,19 @@ impl PartialEq for Type {
 
 pub const FRAG_CC: i32 = 0x66726167; // 'frag'
 
+pub static INT_TYPES_SLICE: &[Type] = &[common_type::int];
+pub static INT2_TYPES_SLICE: &[Type] = &[common_type::int2];
+pub static FLOAT_TYPES_SLICE: &[Type] = &[common_type::float];
+pub static FLOAT3_TYPES_SLICE: &[Type] = &[common_type::float3];
```

Comment:
Rust question. The fact that we use slices here, does it mean that the compiler will likely allocate a single contiguous chunk of memory (array) for all the static and return a slice reference for a portion of it?

Reply:
yep